lib/package_utils.rb

```diff
@@ -23,7 +24,13 @@ def self.incompatible_reason(pkg)
 
   def self.get_url(pkg, build_from_source: false)
     if !build_from_source && pkg.binary_sha256&.key?(ARCH.to_sym)
-      return "https://gitlab.com/api/v4/projects/26210301/packages/generic/#{pkg.name}/#{pkg.version}_#{ARCH}/#{pkg.name}-#{pkg.version}-chromeos-#{ARCH}.#{pkg.binary_compression}"
+      # Fall back to the old method if querying the gitlab API doesn't work for whatever reason.
+      possible_url = get_binary_url(pkg)
+      if possible_url.to_s.empty?
+        return "https://gitlab.com/api/v4/projects/26210301/packages/generic/#{pkg.name}/#{pkg.version}_#{ARCH}/#{pkg.name}-#{pkg.version}-chromeos-#{ARCH}.#{pkg.binary_compression}"
+      else
+        return possible_url
+      end
```

Comment:
Why not move this whole section into `get_binary_url`?

Reply:
I'm handling this here so we only have to state the fallback URL once, rather than three times in `get_binary_url`. 

trying to avoid repeating myself :wink:

Reply:
Who said you needed to return before the end of all the decisions?

Reply:
I don't really understand your point here-- I've got multiple early exits in `get_binary_url` in order to avoid the function crashing, which is why there are multiple returns. This code in `get_url` is here in order to streamline the early exit process, rather than having to return the fallback url in each instance we want to exit early in `get_binary_url`.

Reply:
I added code snippets for moving this section into `get_binary_url`.